src/crimson/os/seastore/seastore_types.cc

```diff
@@ -5,6 +5,15 @@
 
 namespace crimson::os::seastore {
 
+laddr_t get_hint(const ghobject_t &hoid) {
+  uint32_t hash = hoid.hobj.get_hash();
+  if (hoid.shard_id == shard_id_t::NO_SHARD) {
+    return (uint64_t)(hoid.hobj.pool & 0xFF)<<56 | (uint64_t)(hash)<<24;
+  } else {
+    return (uint64_t)(hoid.shard_id.id & 0X7F)<<56 |(uint64_t)(hoid.hobj.pool & 0xFF)<<48 |(uint64_t)(hash)<<16;
```

Comment:
This only left 16 bits of laddr address space for a hash (or PG), is that too small? @athanatos

Reply:
Yeah, we need the full 32 bits for the the hash.

Reply:
all 32 bits of hash is used, I think yingxin means the second branch only 16bits range left for laddr, since the high 48bits is occupied by shard id(8bits), pool id(8 bits) and hash(32bits).

Reply:
Hmm.  It's only a hint, so if a shard,pool,hash combo does exhaust the remaining 16 bits, it'll just roll over into the next pg hash region which should mostly not be in use (or it'll mix with that object, which is also ok if perhaps less performant).  If this becomes a problem, we'll need to consider a wider logical address (96 or 128).